crates/datasources/src/object_store/http.rs

```diff
@@ -94,17 +98,17 @@ impl ObjStoreAccess for HttpStoreAccess {
         }
         // reqwest doesn't check the content length header, instead looks at the contents
         // See: https://github.com/seanmonstar/reqwest/issues/843
-        let len: u64 = res
+        let len = res
             .headers()
             .get("Content-Length")
             .and_then(|v| v.to_str().ok())
-            .and_then(|v| v.parse().ok())
-            .unwrap_or_else(|| res.content_length().unwrap_or(0));
-        if len == 0 {
-            return Err(ObjectStoreSourceError::Static(
+            .and_then(|v| v.parse::<u64>().ok())
+            .or_else(|| res.content_length())
+            // TODO: Download the contents of the file and handle using local
+            // store (maybe outside of HTTP store).
```

Comment:
this is what I did in the `sqlite` from cloud handler, if I understand things correctly. 

Also it seems like, we probably don't want to rely on content-length headers being accurate...